WordPress/Classes/ViewRelated/NUX/SignupUsernameViewController.swift

```diff
@@ -0,0 +1,130 @@
+import SVProgressHUD
+
+class SignupUsernameViewController: NUXViewController {
+    // MARK: - Properties
+    open var currentUsername: String?
+    private var newUsername: String?
+    open var displayName: String?
+
+    // Used to hide/show the Buttom View
+    @IBOutlet weak var buttonContainerViewBottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var buttonContainerHeightConstraint: NSLayoutConstraint!
+
+    override var sourceTag: SupportSourceTag {
+        get {
+            return .wpComCreateSiteUsername
+        }
+    }
+
+    private var usernamesTableViewController: SignupUsernameTableViewController?
+    private var buttonViewController: NUXButtonViewController?
+
+    // MARK: - View
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        addConfirmationWarning()
+    }
+
+    private func configureView() {
+        _ = addHelpButtonToNavController()
+        navigationItem.title = NSLocalizedString("Create New Site", comment: "Create New Site title.")
+        WPStyleGuide.configureColors(for: view, andTableView: nil)
+    }
+
+    private func addConfirmationWarning() {
+        let warningLabel = UILabel()
+        warningLabel.text = NSLocalizedString("Once changed, your old username will no longer be avilable for use.", comment: "Warning shown before user changes their username.")
+        warningLabel.numberOfLines = 0
+        warningLabel.textAlignment = .center
+        warningLabel.textColor = WPStyleGuide.darkGrey()
+        buttonViewController?.stackView?.insertArrangedSubview(warningLabel, at: 0)
+    }
+
+    private func showButtonView(show: Bool, withAnimation: Bool) {
+
+        let duration = withAnimation ? WPAnimationDurationDefault : 0
+
+        UIView.animate(withDuration: duration, animations: {
+            if show {
+                self.buttonContainerViewBottomConstraint.constant = 0
+            }
+            else {
+                // Move the view down double the height to ensure it's off the screen.
+                // i.e. to defy iPhone X bottom gap.
+                self.buttonContainerViewBottomConstraint.constant +=
+                    self.buttonContainerHeightConstraint.constant * 2
+            }
+
+            // Since the Button View uses auto layout, need to call this so the animation works properly.
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+
+    private func changeUsername() {
```

Comment:
The username displayed back on the epilogue is not updated with the newly selected one.